hazelcast/src/main/java/com/hazelcast/map/impl/query/AggregationResultProcessor.java

```diff
@@ -39,6 +43,7 @@ public AggregationResult populateResult(Query query, long resultLimit, Collectio
 
     @Override
     public AggregationResult populateResult(Query query, long resultLimit) {
-        return new AggregationResult(query.getAggregator());
+        Aggregator resultAggregator = serializationService.toObject(serializationService.toData(query.getAggregator()));
```

Comment:
It's in general an insignificant overhead in comparison to the number of QuerableEntry allocations per Query. The aggregator can't be used in a multi-threaded context and has to be cloned for each partition anyway - so there's 271 aggregators per query. This PR fixes the case where it was not cloned in the case of an empty result.